desc/magnetic_fields/_core.py

```diff
@@ -845,10 +1074,16 @@ def B0(self):
     def B0(self, new):
         self._B0 = float(np.squeeze(new))
 
-    def compute_magnetic_field(
-        self, coords, params=None, basis="rpz", source_grid=None, transforms=None
+    def _compute_A_or_B(
```

Comment:
Same comment as I already mentioned for the Coil classes: I think it will be fewer lines of code and easier to understand if you get rid of these methods and just put the logic for either A or B in their respective methods.